dotnet/src/SemanticKernel.Core/Data/TextSearch/VectorStoreTextSearch.cs

```diff
@@ -406,5 +426,357 @@ private async IAsyncEnumerable<string> GetResultsAsStringAsync(IAsyncEnumerable<
         }
     }
 
+    /// <summary>
+    /// Converts a legacy TextSearchFilter to a modern LINQ expression for direct filtering.
+    /// This eliminates the need for obsolete VectorSearchFilter conversion.
+    /// </summary>
+    /// <param name="filter">The legacy TextSearchFilter to convert.</param>
+    /// <returns>A LINQ expression equivalent to the filter, or null if no filter is provided.</returns>
+    [RequiresDynamicCode("Calls Microsoft.SemanticKernel.Data.VectorStoreTextSearch<TRecord>.CreateSingleClauseExpression(FilterClause)")]
+    private static Expression<Func<TRecord, bool>>? ConvertTextSearchFilterToLinq(TextSearchFilter? filter)
+    {
+        if (filter?.FilterClauses == null || !filter.FilterClauses.Any())
+        {
+            return null;
+        }
+
+        var clauses = filter.FilterClauses.ToList();
+
+        // Handle single clause cases first (most common and optimized)
+        if (clauses.Count == 1)
+        {
+            return CreateSingleClauseExpression(clauses[0]);
```

Comment:
We shouldn't have separate methods here for single/multiple, as that cauases code duplication (see CreateSingleClauseExpression and CreateClauseBodyExpression).